modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageListener.java

```diff
@@ -225,6 +225,7 @@ public void onConfigurationCommitted(CommittedConfiguration config) {
     @Override
     public void onSnapshotSave(Path path, Consumer<Throwable> doneClo) {
         storage.snapshot(path)
+                .thenCompose(unused -> storage.flush())
```

Comment:
I think that the storage should call `flush` inside of `snapshot`, we shouldn't manually call them in conjunction